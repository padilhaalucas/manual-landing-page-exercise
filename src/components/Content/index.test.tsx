import React from 'react';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Content from './index';

jest.mock('next/image', () => ({
  __esModule: true,
  default: (props: any) => {
    return <img {...props} />
  },
}));

describe('Components/Content', () => {

  it('should render the main heading', () => {
    render(<Content />);
    const heading = screen.getByText('What we can help with');
    expect(heading).toBeInTheDocument();
  });

  it('should render two ContentItem components', () => {
    render(<Content />);
    const hairLossItem = screen.getByText("Hair loss needn't be irreversible. We can help!");
    const erectileDysfunctionItem = screen.getByText('Erections can be a tricky thing. But no need to feel down!');
    expect(hairLossItem).toBeInTheDocument();
    expect(erectileDysfunctionItem).toBeInTheDocument();
  });

  it('should render the correct subtitles', () => {
    render(<Content />);
    const hairLossSubtitle = screen.getByText('HAIR LOSS');
    const erectileDysfunctionSubtitle = screen.getByText('ERECTILE DYSFUNCTION');
    expect(hairLossSubtitle).toBeInTheDocument();
    expect(erectileDysfunctionSubtitle).toBeInTheDocument();
  });

  it('should render the correct descriptions', () => {
    render(<Content />);
    const description = screen.getAllByText("We're working around the clock to bring you a holistic approach to your wellness. From top to bottom, inside and out.");
    expect(description).toHaveLength(2);
  });

  it('should render images with correct alt text', () => {
    render(<Content />);
    const hairLossImage = screen.getByAltText("Hair loss needn't be irreversible. We can help!");
    const erectileDysfunctionImage = screen.getByAltText('Erections can be a tricky thing. But no need to feel down!');
    expect(hairLossImage).toBeInTheDocument();
    expect(erectileDysfunctionImage).toBeInTheDocument();
  });

  it('should render the correct number for each ContentItem', () => {
    render(<Content />);
    const number1 = screen.getByText('01');
    const number2 = screen.getByText('02');
    expect(number1).toBeInTheDocument();
    expect(number2).toBeInTheDocument();
  });

  it('should match snapshot', () => {
    const tree = renderer.create(<Content />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});