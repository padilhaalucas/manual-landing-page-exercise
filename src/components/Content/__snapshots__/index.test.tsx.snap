// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/Content should match snapshot 1`] = `
<section
  className="px-4 sm:px-8 md:px-16 lg:px-24 xl:px-60 pt-8 pb-12 sm:pb-16 md:pb-20 lg:pb-24 w-full mx-auto bg-white"
>
  <h2
    className="text-2xl sm:text-3xl md:text-4xl font-bold mt-6 sm:mt-8 md:mt-10 mb-20 sm:mb-12 md:mb-16 lg:mb-20 text-center text-dark-green-manual"
  >
    What we can help with
  </h2>
  <div
    className="space-y-16 sm:space-y-20 md:space-y-24"
  >
    <div
      className="flex flex-col lg:flex-row items-center lg:gap-32 relative"
    >
      <div
        className="absolute text-[100px] sm:text-[140px] md:text-[180px] lg:text-[400px] xl:text-[450px] font-medium text-light-blue-manual opacity-20 right-0 lg:right-48 top-[-5%] lg:top-1/2 transform -translate-y-1/2 z-0 tracking-tight"
      >
        01
      </div>
      <div
        className="w-full lg:w-1/2 z-10"
      >
        <img
          alt="Hair loss needn't be irreversible. We can help!"
          height={280}
          layout="responsive"
          src="/assets/hair-loss.jpg"
          width={280}
        />
      </div>
      <div
        className="w-full lg:w-1/2 z-10 bg-transparent py-8 sm:py-12 md:py-16 lg:py-20 lg:px-6"
      >
        <p
          className="text-xs sm:text-sm text-dark-green-manual mb-4 sm:mb-6 font-normal tracking-widest"
        >
          HAIR LOSS
        </p>
        <div
          className="w-full sm:w-5/6 md:w-3/4 lg:w-2/3"
        >
          <h3
            className="text-xl sm:text-2xl font-medium mb-3 sm:mb-4 text-dark-green-manual"
          >
            Hair loss needn't be irreversible. We can help!
          </h3>
          <p
            className="text-dark-green-manual text-sm sm:text-base leading-6 sm:leading-7"
          >
            We're working around the clock to bring you a holistic approach to your wellness. From top to bottom, inside and out.
          </p>
        </div>
      </div>
    </div>
    <div
      className="flex flex-col lg:flex-row-reverse items-center lg:gap-32 relative"
    >
      <div
        className="absolute text-[100px] sm:text-[140px] md:text-[180px] lg:text-[400px] xl:text-[450px] font-medium text-light-blue-manual opacity-20 left-0 lg:left-40 top-[-5%] lg:top-1/2 transform -translate-y-1/2 z-0 tracking-tight"
      >
        02
      </div>
      <div
        className="w-full lg:w-1/2 z-10"
      >
        <img
          alt="Erections can be a tricky thing. But no need to feel down!"
          height={280}
          layout="responsive"
          src="/assets/erectile-dysfunction.jpg"
          width={280}
        />
      </div>
      <div
        className="w-full lg:w-1/2 z-10 bg-transparent py-8 sm:py-12 md:py-16 lg:py-20 lg:px-6"
      >
        <p
          className="text-xs sm:text-sm text-dark-green-manual mb-4 sm:mb-6 font-normal tracking-widest"
        >
          ERECTILE DYSFUNCTION
        </p>
        <div
          className="w-full sm:w-5/6 md:w-3/4 lg:w-2/3"
        >
          <h3
            className="text-xl sm:text-2xl font-medium mb-3 sm:mb-4 text-dark-green-manual"
          >
            Erections can be a tricky thing. But no need to feel down!
          </h3>
          <p
            className="text-dark-green-manual text-sm sm:text-base leading-6 sm:leading-7"
          >
            We're working around the clock to bring you a holistic approach to your wellness. From top to bottom, inside and out.
          </p>
        </div>
      </div>
    </div>
  </div>
</section>
`;
